### DISCUSSION TITLE
[Argo-CD] AutomountServiceAccountToken flag in values file

### DISCUSSION THREAD

**Discussion by ramanNarasimhan77** [argoproj/argo-helm#2899] [General]
I saw that the Argo-CD helm chart has a values file setting to enable/disable automatically mounting the serviceAccountToken (both at pod level and Service account level)

I also came across this PR: https://github.com/argoproj/argo-helm/pull/626 

Given that ArgoCD has Workloads like Controller, Server, ApplicationSet etc which respond to changes on the Kubernetes cluster, isn't it always mandatory to set this to true? Because without the service account token, these processes will not be able to communicate with the API server.

Wont the functionality be impacted if its set to false? In this case, what is the purpose of this flag in the chart?

**Comment by yu-croco** [comment]
Hi @ramanNarasimhan77 , maybe [the discussions in this PR](https://github.com/argoproj/argo-helm/pull/2625) would be the answer against your question ?

**Reply by ramanNarasimhan77** [reply]
Thank you @yu-croco , the discussion in the PR helped me understand this better.
